Sources/Widgets/Widgets3D/LineWidget/behavior.js

```diff
@@ -0,0 +1,261 @@
+import macro from 'vtk.js/Sources/macro';
+
+const MAX_POINTS = 2;
+
+export default function widgetBehavior(publicAPI, model) {
+  model.classHierarchy.push('vtkLineWidgetProp');
+  let isDragging = null;
+  const direction = [0, 0, 0];
+  const coneBehavior = {
+    CONE_HANDLE1_ALONE: 3,
+    CONE_HANDLE2: 2,
+    CONE_HANDLE1: 1,
+  };
+  // --------------------------------------------------------------------------
+  // Display 2D
+  // --------------------------------------------------------------------------
+
+  publicAPI.setDisplayCallback = (callback) =>
+    model.representations[0].setDisplayCallback(callback);
+
+  // --------------------------------------------------------------------------
+  // Interactor events
+  // --------------------------------------------------------------------------
+
+  function ignoreKey(e) {
+    return e.altKey || e.controlKey || e.shiftKey;
+  }
+
+  function updateHandleDirection(behavior, callData) {
+    let bv = behavior;
+    if (bv === 3) {
+      const handle1Pos = model.widgetState.getHandle1List()[0].getOrigin();
+      const WorldMousePos = publicAPI.computeWorldToDisplay(
+        model.renderer,
+        handle1Pos[0],
+        handle1Pos[1],
+        handle1Pos[2]
+      );
+      const mousePos = publicAPI.computeDisplayToWorld(
+        model.renderer,
+        callData.position.x,
+        callData.position.y,
+        WorldMousePos[2]
+      );
+      for (let i = 0; i < 3; i++) {
+        direction[i] =
+          model.widgetState.getHandle1List()[0].getOrigin()[i] - mousePos[i];
+      }
+      bv = 0;
+    } else {
+      const modifier = bv === 1 ? 1 : -1;
+      bv -= 1;
+      const handle1Pos = model.widgetState.getHandle1List()[0].getOrigin();
+      const handle2Pos = model.widgetState.getHandle2List()[0].getOrigin();
+
+      for (let i = 0; i < 3; i++)
+        direction[i] = (handle1Pos[i] - handle2Pos[i]) * modifier;
+    }
+    model.representations[bv].getGlyph().setDirection(direction);
+  }
+
+  function setHandleDirection() {
+    if (model.shapeHandle1 === 'cone' || model.shapeHandle1 === 'arrow') {
+      updateHandleDirection(coneBehavior.CONE_HANDLE1);
+    }
+    if (model.shapeHandle2 === 'cone' || model.shapeHandle2 === 'arrow') {
+      updateHandleDirection(coneBehavior.CONE_HANDLE2);
+    }
+  }
+
+  function setTextPosition() {
```

Comment:
You called this method as a setter but it also returns a value. Can you find a more explicit function name ?